packages/timeline/src/build-code/inject-style-script.js

```diff
@@ -0,0 +1,51 @@
+import crypto from 'crypto'
+
+const replaceMap = {
+  '"': '\\"',
+  "'": "\\'",
+  '`': '\\`',
+}
+
+/**
+ *
+ *
+ * @param {string} string
+ * @returns {string} an escaped CSS string
+ */
+function escape(string) {
+  return String(string).replace(/["'`]/g, matched => {
+    return replaceMap[matched] || matched
+  })
+}
+
+/**
+ *
+ *
+ * @param {string} [styleTag=''] an HTML string containing <style>...</style> tags
+ * @returns {string} an HTML string with JavaScript code that will inject <style> to <head>
+ */
+export function buildInjectStyleScript(styleTag = '') {
+  const styleContent = styleTag
+    .replace(/<\/?style.*?>/g, '') /* trim <style> and </style> tag */
+    .replace(/(?:\r\n|\r|\n)/g, '') /* remove line breaks */
+    .replace(/(\/\*[^*]*\*\/)|(\/\/[^*]*)/g, '') /* remove comments */
```

Comment:
Does this regex replace 
```
/**
 * This is a comment
 */ 
```
to 
empty string ?

Reply:
> Does this regex replace
> 
> ```
> /**
>  * This is a comment
>  */ 
> ```
> 
> to
> empty string ?

I'll update `(\/\*[^*]*\*\/)` with `\/\*[\s\S]*?\*\/` to include the above case.

And I found that the second condition `(\/\/[^*]*)` will replace string like `http://xxx` into `http:`, so I'll add a negative lookbehind to it.

The final script will be:

```js
.replace(/\/\*[\s\S]*?\*\/|(:)?\/\/.*/g, ($0, $1) => ($1 ? $0 : ''))
```


negative lookbehindnegative lookbehind